config/webpack/environment.js

```diff
@@ -1,3 +1,23 @@
-const { environment } = require('@rails/webpacker')
+const { environment } = require('@rails/webpacker');
 
-module.exports = environment
+const customConfig = {
+  resolve: {
+    fallback: {
+      dgram: false,
+      fs: false,
+      net: false,
+      tls: false,
+      child_process: false,
+    },
+  },
+};
+
+environment.config.delete('node.dgram');
```

Comment:
Is this necessary? Why?

Reply:
The default settings throw this error when I execute `bin/webpack` or `rails assets:precompile`
```
configuration.node should be one of these:
   false | object { __dirname?, __filename?, global? }
   -> Include polyfills or mocks for various node stuff.
   Details:
    * configuration.node has an unknown property 'dgram'. These properties are valid:
      object { __dirname?, __filename?, global? }
      -> Options object for node compatibility features.
    * configuration.node has an unknown property 'fs'. These properties are valid:
      object { __dirname?, __filename?, global? }
      -> Options object for node compatibility features.
    * configuration.node has an unknown property 'net'. These properties are valid:
      object { __dirname?, __filename?, global? }
      -> Options object for node compatibility features.
    * configuration.node has an unknown property 'tls'. These properties are valid:
      object { __dirname?, __filename?, global? }
      -> Options object for node compatibility features.
    * configuration.node has an unknown property 'child_process'. These properties are valid:
      object { __dirname?, __filename?, global? }
      -> Options object for node compatibility features.
```
This are the original environment settings:

``` 
Base {
  loaders: ConfigList(7) [
    { key: 'file', value: [Object] },
    { key: 'css', value: [Object] },
    { key: 'sass', value: [Object] },
    { key: 'moduleCss', value: [Object] },
    { key: 'moduleSass', value: [Object] },
    { key: 'nodeModules', value: [Object] },
    { key: 'babel', value: [Object] }
  ],
  plugins: ConfigList(7) [
    { key: 'Environment', value: [EnvironmentPlugin] },
    { key: 'CaseSensitivePaths', value: [CaseSensitivePathsPlugin] },
    { key: 'MiniCssExtract', value: [MiniCssExtractPlugin] },
    { key: 'Manifest', value: [WebpackAssetsManifest] },
    { key: 'Compression', value: [CompressionPlugin] },
    { key: 'Compression Brotli', value: [CompressionPlugin] },
    {
      key: 'OptimizeCSSAssets',
      value: [OptimizeCssAssetsWebpackPlugin]
    }
  ],
  config: ConfigObject {
    mode: 'production',
    output: {
      filename: 'js/[name]-[contenthash].js',
      chunkFilename: 'js/[name]-[contenthash].chunk.js',
      hotUpdateChunkFilename: 'js/[id]-[hash].hot-update.js',
      path: '/home/sangeeky/kommit/rails-training/blog/public/packs',
      publicPath: '/packs/'
    },
    resolve: { extensions: [Array], plugins: [Array] },
    resolveLoader: { modules: [Array], plugins: [Array] },
    node: {
      dgram: 'empty',
      fs: 'empty',
      net: 'empty',
      tls: 'empty',
      child_process: 'empty'
    },
    devtool: 'source-map',
    stats: 'normal',
    bail: true,
    optimization: { minimizer: [Array] }
  },
  entry: ConfigObject {
    application: '/home/sangeeky/kommit/rails-training/blog/app/javascript/packs/application.js'
  },
  resolvedModules: ConfigList(2) [
    {
      key: 'source',
      value: '/home/sangeeky/kommit/rails-training/blog/app/javascript'
    },
    { key: 'node_modules', value: 'node_modules' }
  ]
}
```
In `node` all values are `'empty'`, so I solved with this [Webpack error](https://github.com/webpack/webpack/issues/11649#issuecomment-786705956)

Another answer is downgrade webpack.

Reply:
Solution here [PR#1 webpack error](https://github.com/SanGeeky/rails-training/pull/3#discussion_r614230335)